state-chain/runtime/src/runtime_apis.rs

```diff
@@ -66,6 +69,14 @@ pub struct BoostPoolDepth {
 	pub available_amount: AssetAmount,
 }
 
+#[derive(Encode, Decode, Eq, PartialEq, TypeInfo, Serialize, Deserialize)]
+pub struct BoostPoolDetails {
+	pub available_amount: AssetAmount,
+	pub amounts: BTreeMap<AccountId32, AssetAmount>,
+	pub pending_boosts: BTreeMap<PrewitnessedDepositId, BTreeMap<AccountId32, AssetAmount>>,
```

Comment:
`pending_boosts` might be a bit confusing here. Maybe `deposits_pending_finalization` or `pending_amounts` ?

Reply:
Also, can we adjust it so it follows the specs defined?
```
deposits_pending_finalization: [{
			channel_id,
			source_chain,
			asset,
			booster_shares:[{
				account_id,
				amount
			}]	
		}] // We can derive the total in_use amount from this
```

Reply:
Not sure how I missed this part of the spec on Notion (I find Notion updates hard to track sometimes...). I thought I'd just start by doing the easiest thing, which is exposing all information (which is more or less how it is described in the linear issue), and make any adjustments if requested.

Now that I see the spec, I have questions:
- Why is it necessary to include `source_chain`/`asset`? This RPC for a specific asset already, so it is already known by the caller of this RPC.
- Do we actually want `channel_id` rather than the deposit id (as I'm providing currently)? Channels can have more than one deposit, so I think the deposit id might be necessary to uniquely identify the deposit, and it is more the question whether we want to *additionally* provide channel_id.

Reply:
> Why is it necessary to include source_chain/asset? This RPC for a specific asset already, so it is already known by the caller of this RPC.

Initially, i thought this RPC would have the ability to return the details for all the existing boost pools and the asset/tier params would be optional. The preferred way would be to allow the following:
- No params provided - return the details for all boost pools
- `asset` param provided - return the details for all boost pools for the provided asset
- `asset` & `tier` provided - return the details only for the specific asset/tier pool 

In the first case we will need the asset/source_chain included in the specific object. So lets include it and enable these 3 use cases above.

> Do we actually want channel_id rather than the deposit id (as I'm providing currently)? Channels can have more than one deposit, so I think the deposit id might be necessary to uniquely identify the deposit, and it is more the question whether we want to additionally provide channel_id.

Thats a good point. Lets include the `prewitnessed_deposit_id` here. But lets also keep the other fields for convenience please (`channel_id`, `source_chain`, `asset`). We can do a clean up in the end based on what ended up being used or not. I really can not recall the exact reason why we included `channel_id` here and not `prewitnessed_deposit_id`.

Reply:
> Does this mean that we are only interested in the amounts (the sum of them?) rather than which deposits we are waiting for? Seems like both are important and if you have the deposit id (as I provide currently), you'd be able to derive the amount, but not the other way around.

Its a good point, we might further improve this rpc by including the deposit ids besides the other fields too (`account_id`, `amount`).
I would rather we expose as much details as possible on this RPC. Exposing only the ids means that we have to look elsewhere to get the details related to that deposit.

Reply:
>  Exposing only the ids means that we have to look elsewhere to get the details related to that deposit.

But the amounts are already exposed this very response under `pending_boosts`. I don't think we should return the amounts twice (similar to how you suggested that total_available_amount wasn't necessary).

Reply:
> But lets also keep the other fields for convenience please (channel_id, source_chain, asset). We can do a clean up in the end based on what ended up being used or not. I really can not recall the exact reason why we included channel_id here and not prewitnessed_deposit_id.

There is no problem adding `source_chain` and `asset`, but the reason I asked about `channel_id` specifically is that we don't currently store that information in the boost pool. Let's think whether it is actually required (or might be required in the future), and if so, think of the best way to provide it.

Reply:
> > Exposing only the ids means that we have to look elsewhere to get the details related to that deposit.
> 
> But the amounts are already exposed this very response under `pending_boosts`. I don't think we should return the amounts twice in the same response (similar to how you suggested that total_available_amount wasn't necessary).

You are right. After seeing the snap i have a better understanding of your comments and the current schema looks good!